File: src-tauri/src/ipc/wechat.rs
use tauri::AppHandle;

use crate::error::AppResult;
use crate::services;

use super::off_thread;

/// error 码与前端约定 no-account / recover-failed
#[tauri::command]
pub async fn wechat_acquire_key(app: AppHandle) -> AppResult<services::wechat::AcquireKeyResult> {
    off_thread(move || services::wechat::acquire_key(&app)).await
}

#[tauri::command]
pub async fn wechat_has_key() -> AppResult<bool> {
    off_thread(|| Ok(services::wechat::has_key())).await
}

File: src/platform/wechat.ts
import { call } from './invoke';

export interface AcquireKeyResult {
  ok: boolean;
  wxid?: string;
  /** 稳定错误码：no-account / recover-failed */
  error?: string;
}

export const wechatApi = {
  /** 获取并缓存解密密钥（扫内存，需 Weixin.exe 在运行） */
  acquireKey: () => call<AcquireKeyResult>('wechat_acquire_key'),
  hasKey: () => call<boolean>('wechat_has_key'),
};
